client/lib/feed-stream-store/post-key.js

```diff
@@ -0,0 +1,34 @@
+export function keyForPost( post ) {
+	if ( post.feed_ID && post.feed_item_ID ) {
+		return {
+			feedId: post.feed_ID,
+			postId: post.feed_item_ID
+		};
+	}
+	if ( post.is_external ) {
+		return {
+			feedId: post.feed_ID,
+			postId: post.ID
+		};
+	}
+	return {
+		blogId: post.site_ID,
+		postId: post.ID
+	};
+}
+
+export function keysAreEqual( a, b ) {
```

Comment:
it doesn't care if there's an undefined extra prop.

`_.isEqual( { feedId: 1, postId: 2 }, { feedId: 1, postId: 2, blogId: undefined } )` is false